orttraining/orttraining/test/python/orttraining_test_python_bindings.py

```diff
@@ -233,3 +233,20 @@ def test_export_model_for_inferencing():
         inference_model_file_path = os.path.join(temp_dir, "inference_model.onnx")
         model.export_model_for_inferencing(inference_model_file_path, ["output-0"])
         assert os.path.exists(inference_model_file_path)
+
+
+def test_cuda_execution_provider():
```

Comment:
You could add another test that tests for the failure case.
Set the `CUDA_VISIBLE_DEVICES` to be empty. And then try to use the cuda ep and see if this test fails.
That could be a good way to gauge what happens if user specifies cuda ep but the machine they are running on does not have a cuda device.